maven-plugin/src/main/java/org/openmrs/maven/plugins/utility/SpaInstaller.java

```diff
@@ -83,14 +83,33 @@ private JsonObject convertPropertiesToJSON(Map<String, String> properties) throw
         return result;
     }
 
+    /**
+     * Add a line from the properties file to the new JSON object. Creates nested objects as needed.
+     * If the value contains commas, it is assumed to represent an array of strings. Known array-valued
+     * parameters are always parsed as arrays of strings. e.g.
+     *   `foo=bar,baz` becomes `{ "foo": ["bar", "baz"] }`
+     *   `configUrl=qux` becomes `{ "configUrl": ["qux"] }` because `configUrl` is known to be array-valued
+     *
+     * @param jsonObject the object being constructed
+     * @param propertyKey
+     * @param value
+     */
     private void addPropertyToJSONObject(JsonObject jsonObject, String propertyKey, String value) {
         String[] keys = propertyKey.split("\\.");
         if (keys.length == 1) {
             if (jsonObject.has(keys[0])) {
                 throw new RuntimeException(BAD_SPA_PROPERTIES_MESSAGE +
                         " Encountered this error processing a property containing the key '" + keys[0] + "' and with value " + value);
             }
-            jsonObject.addProperty(keys[0], value);
+            if (value.contains(",") || "configUrls".equals(keys[0])) {
```

Comment:
Are you sure we want to make anything an array if it contains a comma?  Are there use cases where we want this, or possible use cases where we don't?

Reply:
I have no idea, it's just speculative future-proofing. It seems more likely to me that we might add other array-valued parameters than that we will ever add a parameter which can include a `,`. But I could be wrong about that, and am open to the idea of removing this functionality.

Reply:
Yes, that is true, but in that case they array will also need to support single-value arrays like "configUrls", and so will also need to be hard-coded as you had to do with "configUrls", so having the generic behavior doesn't really add value or make sense.

Reply:
Good point. Removing it.